README.md

```diff
@@ -79,10 +85,70 @@ const {
   positionals
 } = parseArgs({ args, options });
 console.log(values, positionals);
-// Prints: [Object: null prototype] { foo: true, bar: 'b' } []ss
+// Prints: [Object: null prototype] { foo: true, bar: 'b' } []
+```
+
+Detailed parse information is available for adding custom behaviours by specifying `tokens: true` in the configuration. The returned tokens have properties describing:
+
+* all tokens
+  * `kind`: { string } One of 'option', 'positional', or 'option-terminator'.
+  * `index`: { number } Index of element in `args` containing token.
+* option tokens
+  * `name`: { string } Long name of option.
+  * `rawName`: { string } How option used in args, like `-f` of `--foo`.
+  * `value`: { string | undefined } Option value specified in args. Undefined for boolean options.
+  * `inlineValue`: { boolean | undefined } Whether option value specified inline, like `--foo=bar`.
+* positional tokens
+  * `value`: { string } Positional value (i.e. `args[index]`).
+* option-terminator token
+
+For example, assuming the following script for `tokens.js`, which uses
+automatic detection of options and no error checking:
+
+```mjs
+import { parseArgs } from 'node:util';
+console.log(parseArgs({ strict: false, tokens: true }));
 ```
 
-`util.parseArgs` is experimental and behavior may change. Join the
+```cjs
+const { parseArgs } = require('node:util');
+console.log(parseArgs({ strict: false, tokens: true }));
+```
+
+This call shows the three kinds of token and their properties:
+
+```console
+$ node tokens.js -d --foo=BAR -- file.txt
+{
+  values: [Object: null prototype] { d: true, foo: 'BAR' },
+  positionals: [ 'file.txt' ],
+  tokens: [
+    {
+      kind: 'option',
+      name: 'd',
+      rawName: '-d',
+      index: 0,
+      value: undefined,
+      inlineValue: undefined
+    },
+    {
+      kind: 'option',
+      name: 'foo',
+      rawName: '--foo',
+      index: 1,
+      value: 'BAR',
+      inlineValue: true
+    },
+    { kind: 'option-terminator', index: 2 },
+    { kind: 'positional', index: 3, value: 'file.txt' }
+  ]
+}
+```
+
+Short option groups like `-abc` expand to a token for each option. The source argument
+for a token is `args[token.index]`.
```

Comment:
```suggestion
for a token is `args[token.index]`. Note that this is not necessarily the same as in the index in the array.
```

Reply:
Updated example, and description. I stumbled a bit with the `not necessarily` and tried different wording tied to the actual example. Hopefully clear.


```
The source argument for a token is `args[token.index]`.
Short option groups like `-xy` expand to a token for each option.
Note that the `x` and `y` tokens above have the same index, since
they come from the same argument.
```